code/modules/overmap/internet/internet_uplink.dm

```diff
@@ -0,0 +1,196 @@
+#define BASE_INTERNET_RANGE 5
+
+var/global/list/internet_uplinks = list()
+/obj/machinery/internet_uplink
+	name = "\improper PLEXUS uplink"
+	desc = "A machine designed to route massive amounts of data to and from PLEXUS receivers in a local area using a miniaturized wormhole."
+	icon = 'icons/obj/machines/internet_uplink.dmi'
+	icon_state = "unpowered"
+	density = 1
+	anchored = 1
+	stat_immune = 0
+
+	construct_state = /decl/machine_construction/default/panel_closed
+	uncreated_component_parts = list(/obj/item/stock_parts/power/terminal)
+	stock_part_presets = list(/decl/stock_part_preset/terminal_setup)
+
+	var/overmap_range = BASE_INTERNET_RANGE
+	var/max_overmap_range = BASE_INTERNET_RANGE
+
+	var/max_temperature = 100 CELSIUS
+	var/power_per_range = 20000
+
+	var/restrict_networks = FALSE		 // Whether or not a network needs to be permitted to use this uplink.
+	var/list/permitted_networks = list() // Network IDs which are permitted to connect through this uplink.
+	var/initial_id_tag = "plexus" 
+
+/obj/machinery/internet_uplink/Initialize()
+	. = ..()
+	global.internet_uplinks += src
+	set_extension(src, /datum/extension/local_network_member, TRUE)
+	if(initial_id_tag)
+		var/datum/extension/local_network_member/uplink_comp = get_extension(src, /datum/extension/local_network_member)
+		uplink_comp.set_tag(null, initial_id_tag)
+	update_range(overmap_range)
+
+/obj/machinery/internet_uplink/Destroy()
+	global.internet_uplinks -= src
+	. = ..()
+
+/obj/machinery/internet_uplink/attackby(var/obj/item/W, var/mob/user)
+	if(IS_MULTITOOL(W))
+		var/datum/extension/local_network_member/uplink = get_extension(src, /datum/extension/local_network_member)
+		uplink.get_new_tag(user)
+		return TRUE
+
+	return ..()
+
+/obj/machinery/internet_uplink/Process()
+	if(stat & (BROKEN|NOPOWER))
+		return
+
+	if(use_power != POWER_USE_ACTIVE)
+		return
+	
+	// Larger ranges not only require more power, but greater cooling.
+	var/inefficiency = clamp(0.3 + (0.1 * (overmap_range - BASE_INTERNET_RANGE)), 0.1, 0.6)
+	
+	var/datum/gas_mixture/env = return_air()
+	if(!istype(env) || env.return_pressure() < 10) // Vacuum heating is insufficient for this machine.
+		take_damage(20, BURN)
```

Comment:
IMO these damage values are too high
Components have maximum health ~30, shielding components have ~60, 
Basic uplink has 240 HP worth of parts (not counting circuitboard) so 10-20 damage per tick would completely destroy every part in couple minutes
Cutting these by 10 would give people some time to fix the cooling failures and repair instead of rebuilding whole machine

Reply:
I went ahead and cut the damage in half, which should hopefully give players a bit more time to notice the problem.